tools/test-proxy/scripts/transition-scripts/generate-assets-json.ps1

```diff
@@ -228,8 +248,42 @@ Function New-Assets-Json-File {
 Function Invoke-ProxyCommand {
   param(
     [string] $TestProxyExe,
-    [string] $CommandArgs
+    [string] $CommandArgs,
+    [string] $TargetDirectory
   )
+  $updatedDirectory = $TargetDirectory.Replace("`\", "/")
+
+  if ($TestProxyExe -eq "docker" -or $TestProxyExe -eq "podman"){
+    $token = $env:GIT_TOKEN
+    $committer = $env:GIT_COMMIT_OWNER
+    $email = $env:GIT_COMMIT_EMAIL
+
+    if (-not $committer) {
+      $committer = & git config --global user.name
+    }
+
+    if (-not $email) {
+      $email = & git config --global user.email
+    }
+
+    if(-not $token -or -not $committer -or -not $email){
+      Write-Error "When running this transition script in `"docker`" or `"podman`" mode, " `
+        + "the environment variables GIT_TOKEN, GIT_COMMIT_OWNER, GIT_COMMIT_EMAIL must be set to reflect the appropriate user. "
+    }
+
+    $targetImage = if ($env:TRANSITION_SCRIPT_DOCKER_TAG) { $env:TRANSITION_SCRIPT_DOCKER_TAG } else { "azsdkengsys.azurecr.io/engsys/test-proxy:latest" }
+
+    $CommandArgs = @(
+      "run --rm --name transition.test.proxy",
+      "-v `"${updatedDirectory}:/srv/testproxy`"",
+      "-e `"GIT_TOKEN=${token}`"",
+      "-e `"GIT_COMMIT_OWNER=${committer}`"",
+      "-e `"GIT_COMMIT_EMAIL=${email}`"",
+      $targetImage,
+      "test-proxy",
+      $CommandArgs
```

Comment:
Ah ok, ignore my earlier comment re: ssh, I didn't realize you added some special handling. I can push an update to allow this to handle ssh mode as well.